### DISCUSSION TITLE
State of subscription

### DISCUSSION THREAD

**Discussion by Sytten** [apollographql/apollo-server#4613] [General]
I would like to start a discussion on the subscriptions. From what I was able to gather the plan for v3 is to remove the package `subscriptions-transport-ws` and bring it in the core so it can support the full request lifecycle and probably harden it since the current solution is quite brittle. Is there a timeline and roadmap for that? Currently I would say subscriptions are barely usable for production use cases.

**Comment by ivanjeremic** [comment]
I was planning to start learning and playing around with subscriptions, what you mean it is not production ready as far as I know it is in the graphql spec and millions are using it in production, or do you mean how apollo did it is not ready for production? 

If that is what you mean can you explain a bit more and also give maybe some alternatives? I know relay but relay is only a client I think, and what about the normal graphql for node this needs to have everything stable regarding subscription?

**Reply by Sytten** [reply]
I am talking about `subscriptions-transport-ws` which has been deprecated since but is still used in apollo server v2. Subscriptions has a concept are in the spec but it doesn't mean an implementation is production ready.
